### PR TITLE
ci: Check for 2 spaces after a period

### PR DESCRIPTION
**Pull request by russellb**
I've received comments on a couple of PRs recently about putting 2
spaces after a period.  This patch will get CI to flag this
automatically.  It also puts the check in an easy command to run
locally, "inv checkpatch", which can be extended for other patch style
checking.

### CODE REVIEW

**Comment by russellb** [comment]
Job runs and fails as expected, because of the grep command that was added that is used to check added lines

File: tasks.py [hunk 1]
```diff
@@ -324,3 +324,21 @@ def _replace(pattern):
...
+        lines = run("git diff $(diff -u <(git rev-list --first-parent HEAD) "
+                                " <(git rev-list --first-parent origin/main) "
+                                " | sed -ne 's/^ //p' | head -1)..HEAD | "
+                                " grep '+.*\.\  '")
+
+        if len(lines.stdout.strip()) > 0:
+            raise Exit(message="ERROR: Found changed lines with 2 spaces "
+                               "after a period.")
+    except UnexpectedExit:
+        # Will exit non-zero if no double-space-after-period lines are found
```
Comment (johananl) [inline tasks.py:343]: Nit: Missing period :-)

**Review by johananl — APPROVED** [review]
LGTM (left one nit).
@russellb if you could rebase and verify this still works as intended, I think we can merge. Thanks!

File: tasks.py (hunk 1 above)
Comment (russellb) [reply tasks.py]: fixed

**Comment by russellb** [comment]
LOL, once I got the job running properly again, it pointed out how I added lines with 2 spaces after a period in a previous PR I already merged.

**Comment by russellb** [comment]
ok, this works for me locally and CI is sorted out now, so I'm going to merge and we'll see if it behaves unexpectedly on any other PRs.

**Pull request merged**